04.ls/ls.rb

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env ruby
+
+@dir = Dir.glob '*'
+# 列数の指定
+column = 3
+num_all_file = @dir.count
+line_num = num_all_file / column
+
+# 行数の計算
+column_count_up = if (num_all_file % column).zero?
+                    line_num
+                  else
+                    line_num + 1
+                  end
+
+# 指定した列数にファイル数が収まらるか判定
+capacity = (num_all_file / column) * column
+if capacity >= num_all_file
+  end_column = column - 1
+else
+  column_count_up = 1
+  end_column = num_all_file
+end
+
+def tab2
+  print(@dir[@column_count], "\t\t")
+end
+
+def new_line
+  print(@dir[@column_count], "\n")
+end
+
+@column_count = 0
+0.upto(num_all_file - 1) do |num|
+  # 最後の列ではない場合
+  if num < end_column
+    tab2
+    @column_count += column_count_up
+  else
+    new_line
+    # 次の行の最後のカラムを計算
+    end_column += column
+    # 次の行の最初のカラムの値を計算
+    @column_count = if num / column < 1
```

Comment:
`num / column < 1` が成立するなら `1 + num / column` は必ず `1` になるように見えますが、この分岐は必要でしょうか？

Reply:
こちらについても処理に不具合がありましたので、修正しております。

> num / column < 1 が成立するなら 1 + num / column は必ず 1 になるように見えますが、この分岐は必要でしょうか？

こちらの処理について例を出すと下記のように6つのファイルを3列で表示したいとします。
```
0	2	4
1	3	5
```

Dir.glob '*'で取得した配列は下記のようになっていますが
```
dir_origin["0", "1", "2", "3", "4", "5"]
```

今回は、変数dirに下記のように要素を並び変えて格納しています。
```
dir_origin["0", "1", "2", "3", "4", "5"]
dir["0", "2", "4", "1", "3", "5"]
```

そして、1行目を表示する時にuptoのnumとget_element変数を(num, get_element)と表した場合、次のようになります。
```
(1, 3)0	(2, 3)2	(3, 3)4
```
そのため、numが(3)になると改行、厳密に言うと指定したカラム数の3の倍数の時に次の行を表示したいので、下記のような処理で次の行の最初の


```suggestion
    # 次の行の最初のカラムの値を計算
    get_element = if num / column > 1
                    num / column
                  else
                    1
                  end
```